### PR TITLE
Add numpoly

### PR DESCRIPTION
**Pull request by jschueller**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/numpoly/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{% set version = "0.1.8" %}
+
+package:
+  name: numpoly
+  version: {{ version }}
+
+source:
+  fn: numpoly-{{ version }}.tar.gz
```
Comment (npavlovikj) [inline recipes/numpoly/meta.yaml]: You can remove `fn` - it is not used in the newer conda-build anymore.

File: recipes/numpoly/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  url: https://github.com/jonathf/numpoly/archive/v{{ version }}.tar.gz
+  sha256: 1c8a7b1343abbe0f407e1f0cbdfbd0270cfab8eef059d9a45f985b076f854534
+
+build:
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
+
+requirements:
+  build:
```
Comment (npavlovikj) [inline recipes/numpoly/meta.yaml]: This should be `host` instead of `build`.

File: recipes/numpoly/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    - python >=3.5
```
Comment (npavlovikj) [inline recipes/numpoly/meta.yaml]: In general, only `pip` and `python` are needed in `host`, and the other dependencies should go in `run`. Also, I don't think there is need of including `setuptools`.

Comment (jschueller) [reply recipes/numpoly/meta.yaml]: ok

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**